enhancements/network/smart-nic-ovn-offload.md

```diff
@@ -0,0 +1,388 @@
+
+Daemonsets of OVN-Kubernetes components other than ovnkube-node-dpu-host
+will be modified with `nodeAffinity` to run on non-DPU hosts.
+
+#### SR-IOV Network Operator
+
+DPU offload will be enabled for nodes in custom MachineConfigPool.
+
+A new optional field `OVNHardwareOffloadConfig` will be added in the
+`SriovNetworkPoolConfigs` CRD spec in SR-IOV Network Operator:
+
+```go
+// SriovNetworkPoolConfigSpec defines the desired state of SriovNetworkPoolConfig
+type SriovOperatorConfigSpec struct {
+	// OvsHardwareOffloadConfig describes the OVS HWOL configuration for selected MachineConfigPool
+	OvsHardwareOffloadConfig OvsHardwareOffloadConfig `json:"ovsHardwareOffloadConfig,omitempty"`
+        // OVNHardwareOffloadConfig describes the OVN HWOL configuration for selected MachineConfigPool
+        OVNHardwareOffloadConfig OVNHardwareOffloadConfig `json:"ovnHardwareOffloadConfig,omitempty"`
+}
+
+type OVNHardwareOffloadConfig struct {
+	// Name is mandatory and must be unique.
+	// Name is the name of MachineConfigPool to be enabled with OVN hardware offload
+	Name string `json:"name,omitempty"`
+}
+```
+
+SR-IOV Network Operator will apply DPU related MachineConfigs by calling Machine
+Config Operator APIs. DPU MachineConfigs include service that reverts default
+OVN-Kubernetes ovs-configuration, and dropin service that disables Open vSwitch
+services (openvswitch, ovs-vswitchd, ovsdb-server).
+
+SR-IOV Network Operator will create `SriovNetworkNodePolicy` CRs
+automatically for nodes in the `MachineConfigPool` selected by
+`OVNHardwareOffloadConfig`. The creation of `SriovNetworkNodePolicy`
+CRs will trigger the provision of SR-IOV devices on those nodes and
+advertise SR-IOV devices (except VF 0) to kubernetes node status as
+extended resource. VF with index 0 will be used by OVN-Kubernetes as
+management port `ovn-k8s-mp0`. SR-IOV Network Operator will ensure
+VF with index 0 is not added to the advertised resource.
+
+SR-IOV Network Operator will configure [DNS Operator placement API](https://github.com/openshift/api/blob/release-4.8/operator/v1/types_dns.go#L64) based on
+MCP specified in `OVNHardwareOffloadConfig` to isolate DNS pods from
+running on DPU hosts (see [Open Questions](#open-questions) for why removing DNS pods from
+DPU hosts).
+
+SR-IOV Network Operator will apply label
+`network.operator.openshift.io/dpu-host` to nodes in the MCP selected by
+`OVNHardwareOffloadConfig` which drives ovn-controller, ovn-ipsec and
+ovs-node daemonset pods away from DPU hosts and allows scheduling of
+ovnkube-node-dpu-host daemonset pods onto DPU hosts.
+
+A new configmap `ovn-management-ports` will be created by SR-IOV Network
+Operator, this configmap is mounted to CNO ovnkube-node-dpu-host daemonset
+to pass node ovn management port information.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ovn-management-ports
+  namespace: openshift-ovn-kubernetes
```

Comment:
I think the SR-IOV operator only has permission to create things in its own namespaces?

Reply:
Yes, it has the permission. Based on previous comment, this part will be handled by User directly. SR-IOV Operator won't create this configmap automatically.